autogluon/utils/tabular/ml/models/tab_transformer/TabTransformer_model.py

```diff
@@ -0,0 +1,320 @@
+            val_data.encode(self.fe)
+        else:
+            val_data=None
+
+        if unlab_data is not None:
+            unlab_data.encode(self.fe)
+
+        return data, val_data, unlab_data
+
+
+
+    def _fit(self, X_train, y_train, X_val=None, y_val=None, X_unlabeled=None, **kwargs):
+        self.set_default_params(y_train)
+
+        train, val, unlab = self.preprocess(X_train, X_val, X_unlabeled)
+   
+        if self.problem_type=='regression':
+            train.targets = torch.FloatTensor(list(y_train))
+            val.targets   = torch.FloatTensor(list(y_val))
+        else:
+            train.targets = torch.LongTensor(list(y_train))
+            val.targets   = torch.LongTensor(list(y_val))
+
+        trainloader, valloader, unlabloader = train.build_loader(), val.build_loader(), unlab.build_loader() if unlab is not None else None
+   
+        self.cat_feat_origin_cards=trainloader.cat_feat_origin_cards
+       
+        self.get_model()
+
+        if X_unlabeled is not None:
+            self.model.fit(unlabloader, valloader, state='pretrain') 
+            self.model.fit(trainloader, valloader, state='finetune')
+        else:
+            self.model.fit(trainloader, valloader)
+
+        """
+        #TODO: If no X_train is passed, but X_unlabeled is, assume that the user wants to 
+        pretrain and save a base model. (see todo list at bottom of script for more info)
+        """
+         
+
+    def predict_proba(self, X, preprocess=False):
+        """
+        X (torch.tensor or pd.dataframe): data for model to give prediction probabilities
+        returns: np.array of k-probabilities for each of the k classes. If k=2 we drop the second probability.
+        """
+        if preprocess or isinstance(X, pd.DataFrame):
+            X, _, _ =self.preprocess(X, fe=self.fe)
+        else:
+            X=X
+     
+        self.model.eval()
+        softmax=nn.Softmax(dim=1)
+        with torch.no_grad():
+            out, _ = self.model(X.cat_data)
+            prob=softmax(out)
+        prob=prob.detach().numpy()
+
+        if prob.shape[1]==2:
+            prob=prob[:,1]
+        
+        return prob
+
+    def save(self, file_prefix="", directory=None, return_filename=False, verbose=True):
+        """
+        file_prefix (str): Appended to beginning of file-name (does not affect directory in file-path).
+        directory (str): if unspecified, use self.path as directory
+        return_filename (bool): return the file-name corresponding to this save
+        """
+        if directory is not None:
+            path = directory+file_prefix
+        else:
+            path = self.path+file_prefix
+
+        params_filepath = path+self.params_file_name
+        
+        os.makedirs(os.path.dirname(path), exist_ok=True)
+
+        if self.model is not None:
+            torch.save(self.model, params_filepath)
```

Comment:
set `self.model = None` after this to avoid pickling the massive neural net weights